src/main/java/SoftDeleteSample.java

```diff
@@ -156,38 +178,45 @@ public static void deletedKeyRecovery() throws Exception {
         String keyToRecoverName = getRandomName("key");
         String keyToPurgeName = getRandomName("key");
 
-        KeyBundle key = keyVaultClient.createKey(vault.vaultUri(), keyToRecoverName, JsonWebKeyType.RSA);
-        System.out.printf("Create key %s.%n", key.keyIdentifier().toString());
 
-        key = keyVaultClient.createKey(vault.vaultUri(), keyToPurgeName, JsonWebKeyType.RSA);
-        System.out.printf("Create key %s.%n", key.keyIdentifier().toString());
+        KeyAsyncClient keyAsyncClient = vault.keyClient();
+
+        KeyVaultKey key = keyAsyncClient.createRsaKey(new CreateRsaKeyOptions(keyToRecoverName).
+                setKeyOperations(KeyOperation.UNWRAP_KEY, KeyOperation.WRAP_KEY,
+                        KeyOperation.VERIFY, KeyOperation.ENCRYPT, KeyOperation.DECRYPT)).block();
+        System.out.printf("Create key %s.%n", key.getId().toString());
 
-        List<KeyItem> keys = keyVaultClient.listKeys(vault.vaultUri());
-        System.out.printf("Keys: %s.%n", Arrays.toString(keys.toArray()));
+        key = keyAsyncClient.createRsaKey(new CreateRsaKeyOptions(keyToPurgeName).
+                setKeyOperations(KeyOperation.UNWRAP_KEY, KeyOperation.WRAP_KEY,
+                        KeyOperation.VERIFY, KeyOperation.ENCRYPT, KeyOperation.DECRYPT)).block();
+        System.out.printf("Create key %s.%n", key.getId());
 
-        DeletedKeyBundle deletedKey = keyVaultClient.deleteKey(vault.vaultUri(), keyToRecoverName);
-        pollOnKeyDeletion(vault.vaultUri(), keyToRecoverName);
+        PagedIterable<Key> keys = vault.keys().list();
+        System.out.printf("Keys: %s.%n", Arrays.toString(keys.stream().collect(Collectors.toList()).toArray()));

```

Comment:
I think it would be better to use our `KeyClient` for all key-related operations.

```suggestion
        PagedIterable<KeyProperties> keyProperties = keyClient.listPropertiesOfKeys();
        System.out.printf("Keys: %s.%n", Arrays.toString(keyProperties.stream().map(KeyProperties::getName).toArray()));
```